### PR TITLE
Add CI and swift-format config

### PR DESCRIPTION
**Pull request by GNMoseke**
This adds basic CI (linting and tests on macOS/ubuntu) for PRs. Some things to keep in mind:

* the .swift-format file here is opinionated based on my personal preferences, I am happy to tweak them based on other maintainers' thoughts
* By default I've set swift to build with -Werror as I think it's a good practice to get into off the starting block. Again, if we forsee unavoidable errors (`@deprecated()` 😡 ) being added, happy to remove those flags
* The macOS tests are running only against the latest stable Xcode - happy to dig into more custom options there if desired

### CODE REVIEW

**Comment by JaapWijnen** [comment]
Can we do the test ci in a separate PR from the formatting perhaps? Since we're still having some concerns around swift format capabilities.

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: Pull Requst
```
Comment (clackary) [inline .github/workflows/pull_request.yml]: typo: Request

**Review by clackary — APPROVED** [review]

**Comment by GNMoseke** [comment]
> Can we do the test ci in a separate PR from the formatting perhaps? Since we're still having some concerns around swift format capabilities.

Sure, sorry have not had time recently to get back to this. Will try and split by monday.

**Comment by GNMoseke** [comment]
See https://github.com/differentiable-swift/swift-differentiation/pull/6#issuecomment-2548241219 - the same changes have been pushed here

bump @JaapWijnen @clackary

File: Sources/OrderedCollectionsDifferentiable/OrderedDictionary+Update.swift [hunk 2]
```diff
@@ -18,7 +18,7 @@ extension OrderedDictionary {
 
 #if canImport(_Differentiation)
 
-extension OrderedDictionary where Value: Differentiable {
+public extension OrderedDictionary where Value: Differentiable {
```
Comment (JaapWijnen) [inline Sources/OrderedCollectionsDifferentiable/OrderedDictionary+Update.swift]: Can we turn this rule off? Prefer having the access level at the definition.

File: Tests/OrderedCollectionsDifferentiableTests/OrderedCollectionsDifferentiableTests.swift [hunk 3]
```diff
@@ -0,0 +1,6 @@
+@testable import OrderedCollectionsDifferentiable
+import Testing
+
+@Test func example() async throws {
```
Comment (JaapWijnen) [inline Tests/OrderedCollectionsDifferentiableTests/OrderedCollectionsDifferentiableTests.swift]: This is added back in by accident I'm assuming?

Comment (GNMoseke) [reply Tests/OrderedCollectionsDifferentiableTests/OrderedCollectionsDifferentiableTests.swift]: yeah, should be gone

File: Sources/OrderedCollectionsDifferentiable/OrderedDictionary+Update.swift (hunk 2 above)
Comment (GNMoseke) [reply Sources/OrderedCollectionsDifferentiable/OrderedDictionary+Update.swift]: sure, fixed

**Pull request merged**